docs/src/npzd_model.md

```diff
@@ -0,0 +1,147 @@
+2. an out-of-place implementation with static matrices and vectors from [StaticArrays.jl](https://juliaarrays.github.io/StaticArrays.jl/stable/).
+
+### Standard out-of-place implementation
+
+Here we create a function to compute the production matrix with return type `Matrix{Float64}`.
+
+```@example NPZD
+using PositiveIntegrators # load ConservativePDSProblem
+
+function prod(u, p, t)
+    N, P, Z, D = u
+
+    p12 = 0.01 * P
+    p13 = 0.01 * Z
+    p14 = 0.003 * D
+    p21 = N / (0.01 + N) * P
+    p32 = 0.5 * (1.0 - exp(-1.21 * P^2)) * Z
+    p42 = 0.05 * P
+    p43 = 0.02 * Z
+
+    return [0.0 p12 p13 p14;
+            p21 0.0 0.0 0.0;
+            0.0 p32 0.0 0.0;
+            0.0 p42 p43 0.0]
+end
+nothing #hide
+```
+The solution of the NPZD model can now be computed as follows.
+```@example NPZD
+u0 = [8.0, 2.0, 1.0, 4.0] # initial values
+tspan = (0.0, 1.0) # time domain
+prob = ConservativePDSProblem(prod, u0, tspan) # create the PDS
+
+sol = solve(prob, MPRK43I(1.0, 0.5))
+
+nothing #hide
+```
+```@example NPZD
+using Plots
+
+plot(sol; label = ["N" "P" "Z" "D"], xguide = "t")
+```
+
+### Using static arrays
+For PDS with a small number of differential equations like the NPZD model the use of static arrays will be more efficient. To create a function which computes the production matrix and returns a static matrix, we only need to add the `@SMatrix` macro.
+
+```@example NPZD
+using PositiveIntegrators # load ConservativePDSProblem
+
+function prod_static(u, p, t)
+    N, P, Z, D = u
+
+    p12 = 0.01 * P
+    p13 = 0.01 * Z
+    p14 = 0.003 * D
+    p21 = N / (0.01 + N) * P
+    p32 = 0.5 * (1.0 - exp(-1.21 * P^2)) * Z
+    p42 = 0.05 * P
+    p43 = 0.02 * Z
+
+    return @SMatrix [0.0 p12 p13 p14;
+                     p21 0.0 0.0 0.0;
+                     0.0 p32 0.0 0.0;
+                     0.0 p42 p43 0.0]
+end
+nothing #hide
+```
+In addition we also want to use a static vector to hold the initial conditions.
+```@example NPZD
+u0_static = @SVector [8.0, 2.0, 1.0, 4.0] # initial values
+tspan = (0.0, 1.0) # time domain
+prob_static = ConservativePDSProblem(prod_static, u0_static, tspan) # create the PDS
+
+sol_static = solve(prob_static, MPRK43I(1.0, 0.5))
+
+nothing #hide
+```
+```@example NPZD
+using Plots
+
+plot(sol_static; label = ["N" "P" "Z" "D"], xguide = "t")
+```
+### Performance comparison
+
+Finally, we use [BenchmarkTools.jl](https://github.com/JuliaCI/BenchmarkTools.jl)
+to show the benefit of using static arrays.
+
+```@example NPZD
+using BenchmarkTools
+@benchmark solve(prob, MPRK43I(1.0, 0.5))
+```
+
+```@example NPZD
+@benchmark solve(prob_static, MPRK43I(1.0, 0.5))
+```
+
+## Package versions
+
+These results were obtained using the following versions.
+```@example LinearAdvection
+using InteractiveUtils
+versioninfo()
+println()
+
+using Pkg
+Pkg.status(["PositiveIntegrators", "SparseArrays", "KLU", "LinearSolve", "OrdinaryDiffEq"],
+           mode=PKGMODE_MANIFEST)
```

Comment:
```suggestion
Pkg.status(["PositiveIntegrators", "StaticArrays", "LinearSolve", "OrdinaryDiffEq"],
           mode=PKGMODE_MANIFEST)
```

Since we're not using sparse arrays here but static arrays